Src/Support/Google.Apis.Auth/OAuth2/ImpersonatedCredential.cs

```diff
@@ -0,0 +1,264 @@
+        internal ImpersonatedCredential(Initializer initializer) : base(initializer)
+        {
+            SourceCredential = initializer.SourceCredential.ThrowIfNull("initializer.SourceCredential");
+            if (!(SourceCredential.UnderlyingCredential is UserCredential || SourceCredential.UnderlyingCredential is ServiceAccountCredential))
+            {
+                throw new InvalidOperationException("The underlying credential of source credential must be UserCredential or ServiceAccountCredential.");
+            }
+            SourceCredential = SourceCredential.CreateScoped(new string[] {GoogleAuthConsts.IamScope});
+            HttpClient.MessageHandler.Credential = SourceCredential.UnderlyingCredential as Http.IHttpExecuteInterceptor;
+            
+            TargetPrincipal = initializer.TargetPrincipal.ThrowIfNullOrEmpty("initializer.TargetPrincipal");
+            Delegates = initializer.Delegates;
+            Scopes = initializer.Scopes;
+            
+            if (initializer.LifetimeInSeconds > 43200)
+            {
+                throw new InvalidOperationException("Lifetime must be less than or equal to 43200.");
+            }
+            if (initializer.LifetimeInSeconds == 0)
+            {
+                LifetimeInSeconds = 3600;
+            }
+            else
+            {
+                LifetimeInSeconds = initializer.LifetimeInSeconds;
+            }
+        }
+
+        /// <inheritdoc/>
+        IGoogleCredential IGoogleCredential.WithQuotaProject(string quotaProject) =>
+            new ImpersonatedCredential(new Initializer(this) { QuotaProject = quotaProject });
+
+        /// <inheritdoc/>
+        public override async Task<bool> RequestAccessTokenAsync(CancellationToken taskCancellationToken)
+        {
+            var bodyJson = new {
+                delegates = this.Delegates,
+                scope = this.Scopes,
+                lifetime = this.LifetimeInSeconds.ToString() + "s"
+            };
+            var body = NewtonsoftJsonSerializer.Instance.Serialize(bodyJson);
+            var content = new StringContent(body, Encoding.UTF8, "application/json");
+            
+            var response = await HttpClient.PostAsync(TokenServerUrl, content, taskCancellationToken).ConfigureAwait(false);
+            var newToken = await TokenResponse.FromHttpResponseAsync(response, Clock, Logger).ConfigureAwait(false);
+            Token = newToken;
+            return true;
+        }
+
+        /// <inheritdoc/>
+        public Task<OidcToken> GetOidcTokenAsync(OidcTokenOptions options, CancellationToken cancellationToken = default)
+        {
+            options.ThrowIfNull(nameof(options));
+            // If at some point some properties are added to OidcToken that depend on the token having been fetched
+            // then initialize the token here.
+            TokenRefreshManager tokenRefreshManager = null;
+            tokenRefreshManager = new TokenRefreshManager(
```

Comment:
This is fine to be all on one line.

Reply:
`tokenRefreshManager` is used in `ct => RefreshOidcTokenAsync(tokenRefreshManager, options, ct)`. Making it one line would throw `tokenRefreshManager` is not defined error.

Reply:
I see. Combined to one line.